gemmforge/thread_policies/factory.py

```diff
@@ -0,0 +1,57 @@
+from gemmforge.vm import VM
+from ..matrix import DenseMatrix
+from .gemm.nvidia import NvidiaGemmThreadPolicy
+from .csa.nvidia import NvidiaCsaThreadPolicy
+from .initializer.nvidia import NvidiaInitializerThreadPolicy
+
+
+class TheadPolicyFactory:
+  ALLOWED_MANUFACTURES = ['nvidia', 'amd', 'sycl']
+
+  def __init__(self):
+    pass
+
+  @classmethod
+  def get_gemm_policy(cls,
+                      vm: VM,
+                      reals_per_op: int,
+                      num_threads: int,
+                      op1: DenseMatrix,
+                      op2: DenseMatrix,
+                      res: DenseMatrix):
+    default_policy = NvidiaGemmThreadPolicy(vm,
+                                            reals_per_op,
+                                            num_threads,
+                                            op1,
+                                            op2,
+                                            res)
+    hw_descr = vm.get_hw_descr()
+    if hw_descr.manufacturer in TheadPolicyFactory.ALLOWED_MANUFACTURES:
+      return default_policy
+    else:
+      raise RuntimeError('unknown manufacturer')
```

Comment:
Hi, @ZaubererHaft. Did you mean this one?

Reply:
Hi @ravil-mobile, yes exactly, with these changes the problem, is fixed, thanks